JPetMC.cpp

```diff
@@ -22,31 +22,33 @@
 #include <G4INCLRandom.hh>
 #include <G4UImanager.hh>
 #include "Info/EventMessenger.h"
-#include <fstream> 
+#include <fstream>
 #include <random>
+#include <TRandom3.h>
 
 void setRandomSeed()
 {
-  long seeds[2];
-  
+  long seed;
+
   std::uniform_int_distribution<long> d(0, LONG_MAX);
   std::random_device rd1;
 
-  seeds[0] = d(rd1);
-  seeds[1] = d(rd1);
-  G4Random::setTheSeeds(seeds);
+  seed = d(rd1);
+
+  G4Random::setTheEngine(new CLHEP::MTwistEngine());
+  G4Random::setTheSeed(seed);
 
-  if (EventMessenger::GetEventMessenger()->SaveSeed()) {
-    std::ofstream file;
-    file.open ("seed", std::ofstream::out | std::ofstream::app);
-    file << seeds[0] << "\n" << seeds[1] << "\n";
-    file.close();
-  }
 }
 
 
 int main (int argc, char** argv)
 {
+  delete gRandom;
```

Comment:
What is this part needed for? Right now, there are three random engines in this code:
- TRandom 
- G4Random
- std::random_device

We should avoid at least adding TRandom, which I think was not used here before.

Reply:
While creating both 3g and 2g decays we are using TGenPhaseSpace which is using gRandom underneath. Without deleting and creating it again with 0 we have gRandom seeded with 4357 which is producing us same decays every single time. TRandom is crucial to have different decays every time. (BTW - exactly the same problem occurs in GeantParser in framework, where we are using default gRandom therefore having the same smearing every time)

We could do without random_device which is used only to generate random, unique seed and instead go for combination of time and PID.

Reply:
Ok, thanks for the explanation. I didn't realize the hidden dependency from TGenPhaseSpace.
However, I think that changing gRandom would better fit inside the `setRandomSeed()` function. What do you think about it?

Reply:
It's a good idea. In the next version I will move it inside that function and replace random_device with combination of time+PID for setting random, unique seed.

Reply:
By the way why just not do:
`gRandom->SetSeed(YOUR_SEED);`
or
`gRandom->SetSeed(0);`
instead of deleting gRandom  ? By default it is TRandom3 anyway.
The only advantage I see is that it is more explicit which genrator you use.